critwall/src/main/java/org/opencadc/critwall/BucketSelector.java

```diff
@@ -67,18 +67,71 @@
 
 package org.opencadc.critwall;
 
+import ca.nrc.cadc.util.HexUtil;
+import ca.nrc.cadc.util.StringUtil;
 import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
 
 /**
  *
  * @author pdowler
  */
 public class BucketSelector {
-    private final Logger log = Logger.getLogger(BucketSelector.class);
-    public final String bucketSelector;
+    private static final Logger log = Logger.getLogger(BucketSelector.class);
+    private final String bucketSelectors;
+    private final int min;
+    private final int max;
+    private final String hexMin;
+    private final String hexMax;
+    private final String hexBuffer = "000";
 
-    BucketSelector(String selectorPrefix) {
-        this.bucketSelector = selectorPrefix;
-        throw new UnsupportedOperationException("TODO");
+    public BucketSelector(String selectors) {
+        InventoryUtil.assertNotNull(BucketSelector.class, "selectors", selectors);
+        this.bucketSelectors = selectors;
+
```

Comment:
In the ctor you will want to split the input, validate the 1 or 2 values, and save. For now, it would be OK to force both values to be the same length and I'm fine with forcing them to be length 1 to start. 

Validate just means that the chars in the hex char set [0-9,a-f]... with odd number of chars you can't use HexUtil without padding. If it was me I'd brush up on the UTF-8 charset order and see if I could use 'char' arithmetic. It is probably better to think about them as strings or even char[ ]

note: 
0-f is 16 buckets, might just be `for (char c = '0', c <= 'f'; c++)`
0 os one bucket (1/16th of the space)
00 is one bucket 00 (1/256 of the space) and this not equivalent to 0
0-15 is 7 buckets: 0, 10, 11, 12, 13, 14, 15
16-f is the the remaining 12+14 = 26 buckets: 16, 17, ... 1f, 2, 3, ... f
 (these last two subdivide bucket 1 unevenly as 10-16 and 17-1f)

Reply:
a quick test of char math for Java produces 0-9 :;<=>?@ABC...abcd. doesn't appear hex-friendly. :(